### ISSUE TITLE
where to test drive this?

### ISSUE THREAD

**Issue by jonleivent** [labwc/labwc#859]
I tried installing on Fedora 37 - got missing symbols.
I tried installing on Fedora Rawhide - also got missing symbols.
I tried building on Debian Bookworm - and initially broke Debian because some of the wlroots dependencies listed include  libelogind-dev and libxcb-util0-dev, which will cause systemd to get removed and brick things.  Leaving those two out, and I still get build problems due to library version mismatches.

I have tried this before on older versions of labwc as well, unsuccessfully.  

For those that claim "just build it - it's easy" - I guess you were lucky to have the planets and library versions all align properly.  I am not so lucky.

I would appreciate it if someone can direct me to a specific tagged (fairly recent) release of libwc that will build or install and run successfully on a specific stable distro such that I don't end up in version hell.  Anyone?

Maybe a flatpak'ed version of labwc, if that's possible?  An appimage?

**Comment by jonleivent** [comment]
@Flrian 

I would prefer having it work on debian 12 eventually.  How much of the wlroots dependencies from the sway wiki did you install?  In other words, if I start from a debian 12 pure install without any desktop at all, what do I need to add, and which version of labwc should I build?  Will 0.6.2 work, or do I need to checkout an earlier one?  Or, did you start from debian 12 already with some desktop installed?  Which?

I will also retry on fedora rawhide - after doing an update there - maybe that will change my luck.

All this is in KVM vms, BWW.

**Comment by Consolatis** [comment]
We are building labwc on Debian Bookworm/Testing as part of our CI pipeline. You can see the dependencies here:
https://github.com/labwc/labwc/blob/master/.github/workflows/build.yml#L72-L81

I am running labwc on Debian Bullseye + random packages from Bookworm/Testing but I would not recommend doing that.

**Comment by jonleivent** [comment]
fedora rawhide already has xorg-x11-server-Xwayland installed.  And labwc is still not working after the latest rawhide update.  When trying to run labwc from a console tty on rawhide, I get (this is hand copied, so may be typos):
```
[user1@fedora ~]$ labwc
00:00:00.000 [../src/config/rcxml.c:899] cannot read (/rc.xml)
libEGL warning: egl: failed to create dri2 screen
00:00:00.108 [EGL] command: eglInitialize, error: EGL_NOT_INITIALIZED (0x3001), message: "DRI2: failed to create screen"
libEGL warning: NEEDS EXTENSION: failing back to kms_swrast
Segmentation fault (core dumped)
```

**Comment by jonleivent** [comment]
@Consolatis 
I'll try that on rawhide after I try debian 12 again.  Only got room for 1 vm at a time on my laptop.
However, if you are right about missing KVM GPU drivers, that should show up on debain 12 as well.

**Comment by jonleivent** [comment]
What `qemu-system-x86_64` options should I be using?

**Comment by jonleivent** [comment]
Following the yaml file, sucessfully built labwc on debian 12.  Thanks again @Consolatis!  And of course, I get similar rcxml.c etc. errors to rawhide.  So I tried gdb labwc, but doing a bt after it crashes says No stack. Bummer.

**Comment by Consolatis** [comment]
The rc.xml error seems to use a wrong path (`/rc.xml`).
It seems your login session is not set up correctly. At least `$HOME` and `$XDG_RUNTIME_DIR` should be set before starting labwc. This is usually done when logging in but you can also set them manually as a work around. labwc should also run without any `rc.xml` file though but `XDG_RUNTIME_DIR` is required for labwc to make the wayland socket available to other applications.

**Comment by jonleivent** [comment]
@Consolatis 

Looks correct to me (again, hand copied, so maybe typos):
```
$ printenv | grep XDG
XDG_SEAT=seat0
XDG_SESSION_TYPE=tty
XDG_SESSION_CLASS=user
XDG_VTNR=1
XDG_SESSION_ID=3
XDG_RUNTIME_DIR=/run/user/1000
$ echo $HOME
/home/user1
```
You mean set `WLR_RENDERER=pixman` and rebuild labwc from scratch?

**Comment by jonleivent** [comment]
Running labwc with `WLR_RENDERER=pixman` doesn't help.  I still get the same rcxml.c error.

Maybe I should use a debian 12 partition on bare metal instead of a KVM?

**Comment by Consolatis** [comment]
You can ignore the rcxml error for now, labwc should run fine with the default settings.
Do you get other errors when starting like `WLR_RENDERER=pixman build/labwc` (from the labwc repository directory) ?

**Comment by jonleivent** [comment]
Oh - after the errors I get "Could not open VT for client".  Permission problems?  The user is in the video group.

**Comment by Consolatis** [comment]
`adduser username tty` might solve this specific error. But it seems something is off regarding seat management. You'll likely get the next error about input devices not being available.

You can try instead
`apt install libseat-dev policykit-1`

It is a while since I set this up so I am not sure right now if that is all that was required.
If you have network on your VM, you could start labwc with debugging enabled (`WLR_RENDERER=pixman build/labwc -d | tee /tmp/labwc.log`), log in via SSH and copy the logfile from /tmp/labwc.log to you host system to paste it here.

**Comment by jonleivent** [comment]
Installing libseat-dev and policykit-1, rebooting and retrying with WLR_RENDERER=pixman worked.  Didn't need to adduser to tty.  Thanks!  That was an exercise in patience.  Character building.

Now to add a taskbar and do some config'ing.

Thanks again!

**Comment by Consolatis** [comment]
We should likely add the policykit dependency (and the WLR_RENDERER=pixman fallback) somewhere to the readme.
I think it is usually available on systems that have at least one desktop environment running but for new systems without any desktop environment that would reduce the time required to get labwc running and prevent troubleshooting. Will create a new issue for that.

I will close this issue but if you have further questions or issues feel free to keep commenting here or create a new issue.

**Comment by jonleivent** [comment]
I have a suggestion.  I think it would be a benefit to new users if you had a downloadable VM image with an initial setup that works without additional config'ing.  With a default theme, menus, taskbar, etc. and a few useful apps.   If you generate something like that as part of your test suite - don't throw it away, post it instead.

**Comment by Consolatis** [comment]
I guess we could have a CI pipeline that automatically creates an Alpine or Void VM image file.
If we'd do that, it should likely happen in a separate repository. The important thing to keep in mind is that labwc is not a desktop environment but just a compositor though.
